### PR TITLE
[TASK] Update with changes from 9.5

### PR DESCRIPTION
**Pull request by susannemoog**

### CODE REVIEW

File: Documentation/9-CrosscuttingConcerns/1-localizing-and-internationalizing-an-extension.rst [hunk 1]
```diff
@@ -537,3 +515,221 @@ the template, after this we had a look at the content of the extension.
...
+
+Changes in the rendering:
+
+1) Setting :php:`Typo3QuerySettings->languageMode` does **not** influence how Extbase queries records anymore.
+   The corresponding TypoScript setting :ts:`config.sys_language_mode` is used by the core
+   to decide what to do when a page is not translated to the given language (display 404, or try page with different language).
+   Users who used to set :php:`Typo3QuerySettings->languageMode` to `strict` should use
+   :php:`Typo3QuerySettings->setLanguageOverlayMode('hideNonTranslated')` to get translated records only.
+
+   The old behavior was confusing, because `languageMode` had different meaning and accepted different
```
Comment (Lefaux) [inline Documentation/9-CrosscuttingConcerns/1-localizing-and-internationalizing-an-extension.rst]: "different meaningS" or "a different meaning"

File: Documentation/9-CrosscuttingConcerns/1-localizing-and-internationalizing-an-extension.rst [hunk 2]
```diff
@@ -537,3 +515,221 @@ the template, after this we had a look at the content of the extension.
...
+   values in TS context and in Extbase context.
+
+2) Setting :php:`Typo3QuerySettings->languageOverlayMode` to :php:`true` makes Extbase fetch records
+   from default language and overlay them with translated values. So e.g. when a record is hidden in
+   the default language, it will not be shown. Also records without translation parents will not be shown.
```
Comment (Lefaux) [inline Documentation/9-CrosscuttingConcerns/1-localizing-and-internationalizing-an-extension.rst]: do we want to weave in "connected mode" here?

File: Documentation/9-CrosscuttingConcerns/1-localizing-and-internationalizing-an-extension.rst [hunk 3]
```diff
@@ -537,3 +515,221 @@ the template, after this we had a look at the content of the extension.
...
+:ts:`config.sys_language_overlay` and :php:`$GLOBALS['TSFE']->sys_language_content`
+(calculated based on :ts:`config.sys_language_uid` and :ts:`config.sys_language_mode`).
+So you need to change :php:`Typo3QuerySettings` manually only if your Extbase code should
+behave different than other `tt_content` rendering.
+
+Setting :php:`setLanguageOverlayMode()` on a query influences **only** fetching of the aggregate root. Relations are always
+fetched with :php:`setLanguageOverlayMode(true)`.
+
+When querying data in translated language, and having :php:`setLanguageOverlayMode(true)`, the relations
+(child objects) are overlaid even if aggregate root is not translated.
```
Comment (Lefaux) [inline Documentation/9-CrosscuttingConcerns/1-localizing-and-internationalizing-an-extension.rst]: "even if **the** aggregate root"

File: Documentation/9-CrosscuttingConcerns/1-localizing-and-internationalizing-an-extension.rst [hunk 4]
```diff
@@ -537,3 +515,221 @@ the template, after this we had a look at the content of the extension.
...
+See :php:`QueryLocalizedDataTest->queryFirst5Posts()`.
+
+Following examples show how to query data in Extbase in different scenarios, independent of the global TS settings:
```
Comment (Lefaux) [inline Documentation/9-CrosscuttingConcerns/1-localizing-and-internationalizing-an-extension.rst]: "**The** following examples"

File: Documentation/9-CrosscuttingConcerns/1-localizing-and-internationalizing-an-extension.rst [hunk 5]
```diff
@@ -537,3 +515,221 @@ the template, after this we had a look at the content of the extension.
...
+
+1) Fetch records from the language uid=1 only, with no overlays.
+   Previously (:ts:`consistentTranslationOverlayHandling = 0`):
+
+   It was not possible.
```
Comment (Lefaux) [inline Documentation/9-CrosscuttingConcerns/1-localizing-and-internationalizing-an-extension.rst]: -It
+This

File: Documentation/9-CrosscuttingConcerns/1-localizing-and-internationalizing-an-extension.rst [hunk 6]
```diff
@@ -537,3 +515,221 @@ the template, after this we had a look at the content of the extension.
...
+   The regular query will return :php:`null` if passed `uid` doesn't match
+   the language set in the :php:`$querySettings->setLanguageUid()` method.
+
+Filtering & sorting
+-------------------
+
+When filtering by aggregate root property like `Post->title`,
+both filtering and sorting takes translated values into account and you will get correct results, also with pagination.
+
+When filtering or ordering by child object property, then Extbase does a left join between aggregate root
```
Comment (Lefaux) [inline Documentation/9-CrosscuttingConcerns/1-localizing-and-internationalizing-an-extension.rst]: between **the** aggregate root

File: Documentation/9-CrosscuttingConcerns/1-localizing-and-internationalizing-an-extension.rst [hunk 7]
```diff
@@ -537,3 +515,221 @@ the template, after this we had a look at the content of the extension.
...
+table and child record table.
+Then the filter is applied as where clause. This means that filtering or ordering by child record property
```
Comment (Lefaux) [inline Documentation/9-CrosscuttingConcerns/1-localizing-and-internationalizing-an-extension.rst]: ordering by **a** child record property

File: Documentation/9-CrosscuttingConcerns/1-localizing-and-internationalizing-an-extension.rst [hunk 8]
```diff
@@ -537,3 +515,221 @@ the template, after this we had a look at the content of the extension.
...
+only takes values from child records which uids are stored in db (in most cases its default language record).
```
Comment (Lefaux) [inline Documentation/9-CrosscuttingConcerns/1-localizing-and-internationalizing-an-extension.rst]: -which
+whose

File: Documentation/9-CrosscuttingConcerns/1-localizing-and-internationalizing-an-extension.rst (hunk 2 above)
Comment (susannemoog) [reply Documentation/9-CrosscuttingConcerns/1-localizing-and-internationalizing-an-extension.rst]: as far as I know "connected mode" is still a backend view only concept without bearing on how records are handled by the API.

**Comment by sypets** [comment]
Merged into master.

**Pull request merged**